components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -191,16 +191,27 @@ public class LegacyJobOrchestrator : IJobOrchestrator {
                     _lock.Wait();
                     var currentFileHash = GetChecksum(_legacyCliModel.PublishedNodesFile);
                     var availableJobs = new ConcurrentQueue<JobProcessingInstructionModel>();
-                    if (currentFileHash != _lastKnownFileHash) {
+                    if (currentFileHash != _lastKnownFileHash || retryCount < 3) {
```

Comment:
This was something I ran into while testing the changes, if we do not have this check and exception is raised in `if` block at line 194, retries do not work as `currentFileHash != lastKnownFileHash` condition is not true anymore due to `_lastKnownFileHash = currentFileHash;` assignment in the `if` block. To make if block aware of retries are in play, `retryCount < 3` was added. Other way possibly to fix this could be to add `_lastKnownFileHash = string.Empty;` in

```
catch (IOException) {
         throw; //pass it thru, to handle retries
}
```

Reply:
No changes made regarding this in the next commit, please feel free to suggest otherwise.

Reply:
OK, got you. Thanks for the fix.